idseq_dag/steps/blast_contigs.py

```diff
@@ -270,4 +413,39 @@ def get_top_m8(orig_m8, blast_top_m8, min_alignment_length):
                     top_bitscore = bitscore
                     top_line = line
             if top_line is not None:
-                top_m8f.write(top_line)
+                top_m8f.write(top_line) # TODO: Unify reranked formats for NT and NR
+
+
+    @staticmethod
+    def get_top_m8_nt(blast_output, blast_top_m8, min_alignment_length, min_pident):
+        '''
+        For each contig Q (query) and reference S (subject), extend the highest-scoring fragment alignment of Q to S with other non-overlapping fragments as far as possible, to maximize cumulative bitscore while avoiding overlap in Q.
+
+        For each Q, output the S with highest agscore(Q, S), which is defined as the number of matching base pairs in all fragments that belong to HSP(Q, S).
```

Comment:
I was initially unsure what an HSP was - can we define briefly in to comments? "HSP = a local alignment with no gaps that achieves one of the highest alignment scores in a given search."

Reply:
HSP is a BLAST-terminology.  "A High-scoring Segment Pair (HSP) is a local alignment with no gaps that achieves one of the highest alignment scores in a given search." It might also be helpful to mention that in BLAST (-outfmt 6) output one HSP alignment for each line, yet for a pair of qseqid-sseqid, there can be multiple HSP alignments, which were sorted by MAX_SCORE. However, on the online BLAST visualization page, they showed all the HSPs.

Reply:
Yeah, I rewrote the docstring of this function to talk about "highest-scoring fragment alignment" instead of HSP, because I wanted to avoid jargon, but then forgot to define the HSP variable :)   Good catch.

Reply:
Btw, I wish I felt more confident in the choice of `agscore`;   however, as it stands, it's another empirical choice to validate with benchmarking.

Reply:
Updated PR with additional comments as suggested here.